android/src/main/java/com/bitmovin/player/reactnative/extensions/ReadableMapExtension.kt

```diff
@@ -1,25 +1,43 @@
 package com.bitmovin.player.reactnative.extensions
 
+import com.facebook.react.bridge.ReadableArray
 import com.facebook.react.bridge.ReadableMap
 
 fun ReadableMap.getBooleanOrNull(
     key: String,
 ): Boolean? = takeIf { hasKey(key) }?.getBoolean(key)
 
-/**
- * Reads the [Boolean] value from the given [ReadableMap] if the [key] is present.
- * Returns the [default] value otherwise.
- */
-fun ReadableMap.getOrDefault(
+inline fun <T> ReadableMap.withDouble(
     key: String,
-    default: Boolean,
-) = if (hasKey(key)) getBoolean(key) else default
+    block: (Double) -> T,
+): T? = takeIf { hasKey(key) }?.getDouble(key)?.let(block)
 
-/**
- * Reads the [String] value from the given [ReadableMap] if the [key] is present.
- * Returns the [default] value otherwise.
- */
-fun ReadableMap.getOrDefault(
+inline fun <T> ReadableMap.withMap(
     key: String,
-    default: String?,
-) = if (hasKey(key)) getString(key) else default
+    block: (ReadableMap) -> T,
+): T? = takeIf { hasKey(key) }?.getMap(key)?.let(block)
+
+inline fun <T> ReadableMap.withInt(
```

Comment:
Nit: I think that the convention is to include the nullability in the name.

```suggestion
inline fun <T> ReadableMap.withIntOrNull(
```

Reply:
That convention is for `getter`. I think this is a bit more complex, and actually, we only take care of the non null int... however, the no-op on null might be not clear enough.